homeassistant/components/ipp/config_flow.py

```diff
@@ -215,6 +215,7 @@ def _show_setup_form(self, errors: dict | None = None) -> ConfigFlowResult:
                     vol.Required(CONF_BASE_PATH, default="/ipp/print"): str,
                     vol.Required(CONF_SSL, default=False): bool,
                     vol.Required(CONF_VERIFY_SSL, default=False): bool,
+                    vol.Required(CONF_PROTO_VERSION, default=(2, 0)): tuple[int, int],
```

Comment:
would need to see how config flow handles tuples, not sure ive ever tried

Reply:
indeed, could revert this back to bool (checkmark?) for the UI

Reply:
i think that is a sane option or maybe a dropdown backed by enum of the three common versions (1.1/2.0/2.1) but it would still involve some translations to tuple since itd need to be a string value if im not mistaken for UI purposes.

one of the specs regarding IPP Everywhere is that if you provide discovery info, you must support 2.0/2.1. we also dont use any features of 2.1 so 2.0 is what has been used since the start as it exposes marker/toner levels. so this option really only covers 1.1 from a HA standpoint with manual standpoint.

Reply:
![2024-05-18_09-47-46](https://github.com/home-assistant/core/assets/209757/8e733d1d-fba2-43f8-af0e-e48588e319cf)

Reply:
maybe just specify in the checkbox that legacy is 1.1 or just "Use IPP v1.1 for compatibility with older printers" vs "legacy"

Reply:
Opting for the easy way with just a bool checkbox for now. String updated accordingly, please take another look :)